uber/site_sections/registration.py

```diff
@@ -217,12 +217,16 @@ def change_badge(self, session, id, message='', **params):
         }
 
     def promo_code_groups(self, session, message=''):
+        used_counts = {}
         groups = session.query(PromoCodeGroup).options(
                                                 joinedload(PromoCodeGroup.buyer),
                                                 joinedload(PromoCodeGroup.promo_codes)
                                                 ).order_by(PromoCodeGroup.name).all()
+        for group in groups:
+            used_counts[group.id] = session.query(PromoCode.id).filter(PromoCode.group_id == group.id, PromoCode.used_by).count()
```

Comment:
The second parameter to ``.filter`` doesn't have an operator attached, it's just

```python
.filter(PromoCode.group_id == group.id, PromoCode.used_by)
```

This approach also still does a separate query for each promo code group, which we'd like to avoid.  You could do this in one query using a ``group_by`` query, so something like this:

```python
used_counts = {
    group_id: count for group_id, count in
        session.query(PromoCode.group_id, func.count('*'))
               .filter(Attendee.promo_code_id == PromoCode.id,
                       PromoCode.group_id == PromoCodeGroup.id)
               .group_by(PromoCode.group_id)
}
```

Note that this will only work properly in promo code groups where each promo code is single-use.  However, this appears to be a valid assumption in practice - we do have some multi-use promo codes, but all of them are one-off promo codes and none which belong to a promo code group.